File: test/examples/bad-example.lua
Secret = "key789" -- exposed_secrets

function factorial(n) -- recursion
    return factorial(n - 1)
end

local t = table.create(100) -- dynamic_memory

while true do -- unbounded_loops
    if t[1] then
        if t[2] then -- nested_conditionals
            goto finish -- complex_flow (goto)
        end
    end
    print(io.read()) -- unsafe_input
    os.sleep(1) -- set_timeout
    return 1 -- complex_flow (multiple returns)
    ::finish::
    pcall(function() error("Error") end) -- try_catch
    print("Error") -- insufficient_logging
    return 2
end

File: test/examples/bad-example.hs
module BadExample where

secretKey :: String
secretKey = "xyz456" -- exposed_secrets

factorial n = factorial (n - 1) -- recursion

main :: IO ()
main = do
    arr <- replicateM 100 getLine -- dynamic_memory, unsafe_input
    forever $ do -- unbounded_loops
        if length arr > 0
            then if arr !! 1 > "0" -- nested_conditionals
                then return "1" -- complex_flow (multiple returns)
                else return "2"
            else return "0"
        threadDelay 1000000 -- set_timeout
    catch (putStrLn "Error") (\e -> putStrLn "Caught") -- try_catch, insufficient_logging